modules/opsgenie.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+import json
+import logging
+import requests
+
+from elastalert.alerts import Alerter
+from elastalert.alerts import BasicMatchString
+
+
+class OpsGenieAlerter(Alerter):
+    '''Sends a http request to the OpsGenie API to signal for an alert'''
+    required_options = frozenset(['opsgenie_key', 'opsgenie_account', 'opsgenie_recipients'])
+
+    def __init__(self, *args):
+        super(OpsGenieAlerter, self).__init__(*args)
+
+        self.account = self.rule.get('opsgenie_account', 'genie')
+        self.api_key = self.rule.get('opsgenie_key', 'key')
+        self.recipients = self.rule.get('opsgenie_recipients', ['genies'])
+        self.to_addr = self.rule.get('opsgenie_addr', 'example.opsgenie.net')
+
+    def alert(self, matches):
+        body = ''
+        for match in matches:
+            body += str(BasicMatchString(self.rule, match))
+            # Separate text of aggregated alerts with dashes
+            if len(matches) > 1:
+                body += '\n----------------------------------------\n'
+
+        post = {}
+        post['apiKey'] = self.api_key
+        post['message'] = self.create_default_title(matches)
+        post['recipients'] = self.recipients
+        post['description'] = body
+        post['source'] = 'ElastAlert'
+        post['tags'] = ['ElastAlert', self.rule['name']]
+        logging.debug(json.dumps(post))
+
+        try:
+            r = requests.post(self.to_addr, json=post)
+
+            logging.debug('request response: {0}'.format(r))
+            if r.status_code != 200:
+                logging.error("Error sending alert request to OpsGenie! {0}".format(r.json()))
+                r.raise_for_status()
+            logging.info("Alert sent to OpsGenie")
+        except Exception as err:
+            logging.error("Error sending alert to OpsGenie: {0}".format(err))
+            return
```

Comment:
Can this re-raise the exception as util.EAException? That will cause it to be caught in ElastAlert and 
1. Upload an elastalert_error document to Elasticsearch
2. Save the alert as unsent, where it can be retried
3. Logged eventually here https://github.com/Yelp/elastalert/blob/master/elastalert/elastalert.py#L1150

So I guess if you reraise as EAException you can also remove the logging.error from here.